avail-server/src/main/java/com/avail/server/io/WebSocketChannel.kt

```diff
@@ -56,15 +58,44 @@ import java.nio.channels.AsynchronousSocketChannel
  *   The [WebSocketAdapter].
  * @param transport
  *   The [channel][AsynchronousSocketChannel].
+ * @param heartbeatInterval
+ *   The time in milliseconds between each heartbeat request made by the server
+ *   to the client.
+ * @param heartbeatTimeout
+ *   The amount of time in milliseconds the heartbeat will fail if a heartbeat
+ *   is not received from the client by the server.
+ * @param heartbeatFailureThreshold
+ *   The number of consecutive times the `heartbeatTimeout` is allowed to be
+ *   reached before disconnecting the client.
+ * @param onChannelCloseAction
+ *   A function that accepts a [DisconnectReason] and the underlying
+ *   [channel][AbstractTransportChannel] and answers `Unit` that is to be
+ *   called when the input channel is closed.
  */
 internal class WebSocketChannel constructor(
 	override val adapter: WebSocketAdapter,
-	override val transport: AsynchronousSocketChannel)
+	override val transport: AsynchronousSocketChannel,
+	heartbeatFailureThreshold: Int,
+	heartbeatInterval: Long,
+	heartbeatTimeout: Long,
+	onChannelCloseAction:
+		(DisconnectReason, AbstractTransportChannel<AsynchronousSocketChannel>)
+			-> Unit)
```

Comment:
We probably want to default these in the same way that you did for `WebSocketAdapter`.

Reply:
You can use the same constants that you will introduce for the default arguments here.

Reply:
Actually, I specifically didn't do this as I believe we are treating the interface to the server for an AvailServer implementation is to be through the WebSocketAdapter, not WebSocketChannel. I don't believe a user using the server in their implementation would ever change the WebSocketAdapter code where a WebSocketChannel is created. For consistency and accessibility, I added these defaults at the WebSocketAdapter level as this is where you would override them with your own inputs. Additionally, with the addition of a configuration, you would also want to control these settings through configuration, not anywhere else.